plutus-chain-index-core/src/Plutus/ChainIndex/Handlers.hs

```diff
@@ -398,9 +399,16 @@ appendBlocks ::
     , Member (BeamEffect Sqlite) effs
     , Member (LogMsg ChainIndexLog) effs
     )
-    => [ChainSyncBlock] -> Eff effs ()
-appendBlocks [] = pure ()
-appendBlocks blocks = do
+    => L.Slot -> [ChainSyncBlock] -> Eff effs ()
+appendBlocks newSlot [] = do
```

Comment:
Is `appendBlocks :: Either L.Slot (List1 ChainSyncBlock) -> Eff effs ()` an overkill here? It would avoid expecting that the slot is taken into account when some blocks are passed.

Reply:
Thanks for prodding. After looking at it again the real fix was somewhere else.